user-service/src/main/java/uk/gov/gchq/palisade/service/user/impl/AbstractLdapUserService.java

```diff
@@ -103,7 +103,6 @@ public AbstractLdapUserService(final String ldapConfigPath)
     @Override
     public User getUser(final UserId userId) {
         requireNonNull(userId, "userId has not been set");
-        requireNonNull(userId.getId(), "userId has not been set");
```

Comment:
Shift over to the use of Optional.ofNullable

Reply:
This value cannot be `null` so using `Optional.ofNullable` would not work. I will add the `userId.getId()` check back in though.